aws/resource_aws_spot_fleet_request_test.go

```diff
@@ -525,6 +621,12 @@ func TestAccAWSSpotFleetRequest_diversifiedAllocation(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "allocation_strategy", "diversified"),
 				),
 			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"wait_for_fulfillment"},
```

Comment:
@DrFaust92 this test is an interesting one b/c the `diversified` allocation_strategy implies that the `instance_pools_to_use_count` a user provides will be ignored,  by the provider (as the count is set to 1) and according to the AWS [sdk](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_SpotFleetRequestConfigData.html), so it explains the test failure here in the `ImportStateVerify` step

Reply:
perhaps a schema change is in order to address the fact that the default value of `instance_pools_to_use_count` is technically not valid for cases where `allocation_strategy` is not `lowest-price` @bflad?

Reply:
..though to avoid too much change and since the field is set with the `ForceNew` attribute, a possible work around could be to explicitly set `instance_pools_to_use_count` w/in `resourceAwsSpotFleetRequestRead` (for cases where the config object returns nil) to match the default expected value of 1

Reply:
There are a few options in this scenario:

* Add `instance_pools_to_use_count` to the `ImportStateVerifyIgnore`
* Add `d.Set("instance_pools_to_use_count", 1)` in the import state function
* Adjust the schema for `instance_pools_to_use_count` from `Default: 1` to `Computed: true` to denote its multiple defaults (this however will require additional adjustments in the `Create` logic to check the allocation strategy in addition to the `!= 1`)

The first option would just make the tests pass for the sake of passing (still representing an operator user experience problem after import). The second option would technically be a no-op change to the resource, since that would match the implicit `Create` behavior where the attribute `Default` is passed through to the state because `d.Set()` in `Read` is not overwriting the value. The third option would technically be the most correct given the API descriptions, however it would prevent that attribute from _always_ being set as it is today, which represents a slight breaking change if any configurations are dependent on that behavior in attribute references.

I think I lean towards option two as the safest choice to implement import to match the existing behavior and a separate followup issue can be added to track fixing the attribute schema and its handling in `Create`, potentially saved for a future major version (unless its actually causing issues of course!).

Reply:
ill address this with the second option as proposed